### PR TITLE
Feature/fga/avoid multiple sync loops

### PR DESCRIPTION
**Pull request by ganfra**
Attempts to avoid multiple sync loops because of https://github.com/matrix-org/matrix-rust-sdk/issues/2247

### CODE REVIEW

**Review by jmartinesp — APPROVED** [review]
LGTM!

File: libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt [hunk 1]
```diff
@@ -49,6 +56,10 @@ class RustSyncService(
         roomListService
             .stateFlow()
             .map(RoomListServiceState::toSyncState)
+            .onEach { state ->
+                Timber.v("Sync state=$state")
+                isSyncing.set(state == SyncState.Syncing)
+            }
```
Comment (bnjbvr) [inline libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt:62]: Is this properly mapping `RoomListServiceState::{Init, SettingUp}` to `Syncing`? If not, this is wrong, and may cause a `true -> false` transition (if the room list service was explicitly started, and then it's moving on to `SettingUp`)

Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt:62]:
```kotlin
internal fun RoomListServiceState.toSyncState(): SyncState {
    return when (this) {
        RoomListServiceState.INIT,
        RoomListServiceState.SETTING_UP -> SyncState.Idle
        RoomListServiceState.RUNNING -> SyncState.Syncing
        RoomListServiceState.ERROR -> SyncState.InError
        RoomListServiceState.TERMINATED -> SyncState.Terminated
    }
}
```

Only `Running` is mapped to `Syncing`, so I *think* it should be fine?

Comment (jmartinesp) [reply libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/sync/RustSyncService.kt:62]: > you start the sliding sync, so the atomic gets set to true

My bad, I didn't realise we were still using `compareAndSet` there. @ganfra maybe we should re-think this?

**Pull request merged**